application-task-ui/src/main/resources/TaskManager/GanttMacro/Macro.xml

```diff
@@ -0,0 +1,2059 @@
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>long|short|default|forbid</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </cache>
+      <code>
+        <contenttype>PureText</contenttype>
+        <disabled>0</disabled>
+        <editor>PureText</editor>
+        <name>code</name>
+        <number>2</number>
+        <prettyName>Code</prettyName>
+        <rows>20</rows>
+        <size>50</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </code>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>1</number>
+        <prettyName>Name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parse>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>require(['jquery', frappe_js_path], function(jQuery){
+  // Add custom popup and the custom svg markup when rendering Gantt
+  if(Gantt.prototype['_render'] === undefined) {
+    function setupProgressPopup(ganttObj) {
+      let ganttDiv = ganttObj.$svg.parentNode.parentNode;
+      let popup = jQuery('div.gantt-popup', ganttDiv);
+      if(popup.length &gt; 0) {
+        popup = popup[0];
+      } else {
+        popup = document.createElement('div');
+        popup.innerHTML = `
+          &lt;span class='popuptext'&gt;Progress: &lt;span id="gantt-popup-progress-number"&gt;100&lt;/span&gt;%&lt;/span&gt;
+        `;
+        popup.classList.add('gantt-popup');
+        ganttDiv.appendChild(popup);
+      }
+      // Set up popup event listeners
+      let progressChangeObserver;
+      let _hoverTimer;
+      let is_resizing_progress = false;
+
+      function showPopup(barElem) {
+        if (is_resizing_progress) return;
```

Comment:
It's better to always use `{}`, even if it's just a line
For example, someone could come and write another line after `return`, without realizing it will not be included in the `if` statement